deploy/charts/harvester/templates/_helpers.tpl

```diff
@@ -123,3 +123,35 @@ Get Support-bundle-kit image environment for updating the default values per cur
 {{- end -}}
 {{- end -}}
 {{- end }}
+
+{{/*
+Get a dynamic storageclass.kubernetes.io/is-default-class value for harvester-longhorn storageclass.
+*/}}
+{{- define "harvester.annotations.defaultstorageclass" -}}
+{{- if .Values.storageClass.defaultStorageClass }}
+ {{- $allscs := (lookup "storage.k8s.io/v1" "StorageClass" "" "") -}}
+ {{- if eq (len $allscs ) 0 -}}
+storageclass.kubernetes.io/is-default-class: "true"
+ {{- else -}}
+ {{- $scname := "" -}}
+ {{- range $index, $cursc := $allscs.items -}}
+ {{- range $k, $v := $cursc.metadata.annotations -}}
+ {{- if eq $k "storageclass.kubernetes.io/is-default-class" -}}
+ {{- if eq $v "true" -}}
+ {{- $scname = $cursc.metadata.name -}}
+ {{- end -}}
+ {{- end -}}
+ {{- end -}}
+ {{- end -}}
+ {{- if eq $scname "" }}
+storageclass.kubernetes.io/is-default-class: "true"
+ {{- else if eq $scname "harvester-longhorn" -}}
```

Comment:
do we need to check if sc is `harvester-longhorn`? As long as there is another sc which is the default sc we should just let it be the default sc

Reply:
the check is is essential, it identifies who has the `default` anno, if it is still `harvester-longhorn` then keep it, if it is others, `harvester-longhorn` needs to set `default` as `false` (not the managedchart input) to avoid denied by webhook